### PR TITLE
feat(side menu): collapse button improved & header updated in demos

### PR DESCRIPTION
**Pull request by gaonita**
**Describe pull-request**  
- improvement side menu collapse button in Angular demos(sticky position on the bottom of the screen)
- update header(v2.0.0) in all Angular demos and React demo 

**Solving issue**  

Fixes: [AB#1371](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1371)

**How to test**  

1. Go to Angular(11, 12,13) demo and React demo
2. Check header and side menu collapse button the look and behaviour

### CODE REVIEW

**Review by areblk — CHANGES_REQUESTED** [review]
### Angular demos request
1. **Use the same kind of header in all Angular demos.** Header code now differs between angular11+12 and angular13 demo. The later uses the header code and css classes found on digitaldesign.scania.com and in stories.js-file. A11+12 header stays on top when scrolling, A13 header doesn't.
2. **Make collapse button sticky on bottom screen in all demos.** Collapse button sticky on bottom of screen only in A13. In A11+12 it is sticky at the bottom of the _page_ hiding it from users when the page is long.
3. **Fix icon bump expanded/collapsed dropdown in A11+12**. Previous fix for dropdown in side menu bump on icon not in this branch for A11+12.
4. **Clarify expected behavior for mobileView in A13-demo**. When opening mobile menu, it sticks to the page when I scroll down but header moves up out of view. This creates a weird gap between side menu and header that doesn't look nice. What's the expected behavior on mobile? It feels natural that it would close when clicking outside of side menu or start scrolling, i.e. interacting with page. (In comparison, A11+12 that uses a different header, hamburger icon turns to a X when mobile menu is visible. On scroll, header stays on top.)

**Comment by gaonita** [comment]
> ### Angular demos request
> 1. **Use the same kind of header in all Angular demos.** Header code now differs between angular11+12 and angular13 demo. The later uses the header code and css classes found on digitaldesign.scania.com and in stories.js-file. A11+12 header stays on top when scrolling, A13 header doesn't.
> 2. **Make collapse button sticky on bottom screen in all demos.** Collapse button sticky on bottom of screen only in A13. In A11+12 it is sticky at the bottom of the _page_ hiding it from users when the page is long.
> 3. **Fix icon bump expanded/collapsed dropdown in A11+12**. Previous fix for dropdown in side menu bump on icon not in this branch for A11+12.
> 4. **Clarify expected behavior for mobileView in A13-demo**. When opening mobile menu, it sticks to the page when I scroll down but header moves up out of view. This creates a weird gap between side menu and header that doesn't look nice. What's the expected behavior on mobile? It feels natural that it would close when clicking outside of side menu or start scrolling, i.e. interacting with page. (In comparison, A11+12 that uses a different header, hamburger icon turns to a X when mobile menu is visible. On scroll, header stays on top.)
> 
> ### React demo notes
> * Icon dumping when dropdown expants not fixed
> * Header and collapse button is sticky, but side menu scrolls with page.
> * Uses the old header code

@areblk it seems like you are not seeing the change because you are in old version. Make sure you are getting the correct version of package. Let me help you to get the right version. maybe I can tell you in the check in today

File: demo/angular/angular12/package.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
     "@angular-devkit/build-angular": "~12.2.1",
     "@angular/cli": "~12.2.1",
     "@angular/compiler-cli": "~12.2.0",
-    "@scania/components": "^1.14.0",
-    "@scania/icons": "^1.1.1",
-    "@scania/theme-light": "^3.6.2",
+    "@scania/components": "latest",
```
Comment (theJohnnyMe) [inline demo/angular/angular12/package.json:30]: Nice!

**Comment by areblk** [comment]
> > ### Angular demos request
> > 
> > 1. **Use the same kind of header in all Angular demos.** Header code now differs between angular11+12 and angular13 demo. The later uses the header code and css classes found on digitaldesign.scania.com and in stories.js-file. A11+12 header stays on top when scrolling, A13 header doesn't.
> > 2. **Make collapse button sticky on bottom screen in all demos.** Collapse button sticky on bottom of screen only in A13. In A11+12 it is sticky at the bottom of the _page_ hiding it from users when the page is long.
> > 3. **Fix icon bump expanded/collapsed dropdown in A11+12**. Previous fix for dropdown in side menu bump on icon not in this branch for A11+12.
> > 4. **Clarify expected behavior for mobileView in A13-demo**. When opening mobile menu, it sticks to the page when I scroll down but header moves up out of view. This creates a weird gap between side menu and header that doesn't look nice. What's the expected behavior on mobile? It feels natural that it would close when clicking outside of side menu or start scrolling, i.e. interacting with page. (In comparison, A11+12 that uses a different header, hamburger icon turns to a X when mobile menu is visible. On scroll, header stays on top.)
> > 
> > ### React demo notes
> > 
> > 5. Icon dumping when dropdown expants not fixed
> > 6. Header and collapse button is sticky, but side menu scrolls with page.
> > 7. Uses the old header code
> 
> @areblk it seems like you are not seeing the change because you are in old version. Make sure you are getting the correct version of package. Let me help you to get the right version. maybe I can tell you in the check in today

This fixes 1, 2, 3, 5, 7 and half of 6.

Still concerned about 
- Number 4. Where side menu is fixed on mobile but header moves up when scrolling, not able to close menu unless I scroll back up and click hamburger icon. 
- Collapse button is sticky, but side menu scrolls and header scrolls with page.

Maybe this is not a problem with out components per se (because we don't offer js-behavior with them?), but I think it looks bad in the demos.

image

image

**Comment by gaonita** [comment]
Good catch again! @areblk, it does look bad. @theJohnnyMe could you look into the header? was it supposed to be sticky?
I don't think it is a blocker though, since it technically didn't break anything. And the plan was to update demos with new header after this.

**Comment by theJohnnyMe** [comment]
> 
> 
> Good catch again! @areblk, it does look bad. @theJohnnyMe could you look into the header? was it supposed to be sticky? I don't think it is a blocker though, since it technically didn't break anything. And the plan was to update demos with new header after this.

I see the issue. It is hard t predict will people use it as fixed one or it should scroll. However, in the top parent HTML element of Header, next to the sdds-nav class add two mor classes: "sdds-u-fixed sdds-u-top0". I think it should fix the issue.

**Comment by gaonita** [comment]
Added those classes and it works perfectly. Thanks @theJohnnyMe :) please confirm it again @areblk

**Review by areblk — APPROVED** [review]
_Side menu still moves with page, but I guess this is intentional._

Confirming that header is sticky on mobile and desktop with React, Angular 11, Angular 12 and Angular 13.

_Weird bug with Edge and Angular13._ Header sticky on "mobile" initially, but when I open side menu, scroll down, and then close side menu, header isn't sticky anymore. Doesn't go away when I run `npm i` and `npm start` or clean cache. Perhaps something wrong with Edge simulator for mobile devices?

File: components/src/patterns/side-menu/side-menu.stories.js [hunk 2]
```diff
@@ -48,7 +48,8 @@ const Template = (args) => {
     .sdds-navbar-menu-popover {
       display:none;
     }
-    .popover-menu-parent:hover{
+    /*TODO: remove lines 51-55 after release*/
```
Comment (theJohnnyMe) [inline components/src/patterns/side-menu/side-menu.stories.js:51]: Nice! (See if you can clean up one from previous release, not in this PR but another one)

File: demo/angular/angular12/src/app/components/side-menu/side-menu.component.html [hunk 3]
```diff
@@ -57,10 +57,10 @@
           <div class="sdds-navbar-menu-item-dropdown"
                style="height: 100%"
                (mouseenter)="showDropdownPopover=true">
-            <div class="sdds-navbar-menu-item-dropdown-parent">
-              <a class="sdds-navbar-menu-item-link popover-title">
+            <div class="sdds-navbar-menu-item-dropdown-parent popover-dropdown-menu-parent">
+              <div class="sdds-navbar-menu-item-link popover-title">
```
Comment (theJohnnyMe) [inline demo/angular/angular12/src/app/components/side-menu/side-menu.component.html:61]: Nice!

File: demo/react/src/components/header/Header.css [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-.sdds-navbar-menu-list .sdds-navbar-menu-item.active,
-.sdds-navbar-menu-list .sdds-navbar-menu-item:active,
-.sdds-navbar-menu-list .sdds-navbar-menu-item:visited {
-  --sdds-link: var(--sdds-white);
-  --sdds-link-visited: var(--sdds-white);
-  --sdds-link-focus: var(--sdds-white);
-  --sdds-link-focus-border: none;
-  --sdds-link-text-decoration: none;
-}
+/*.sdds-navbar-menu-list .sdds-navbar-menu-item,*/
```
Comment (theJohnnyMe) [inline demo/react/src/components/header/Header.css:1]: If it is not needed anymore - drop it.

File: demo/react/src/index.scss [hunk 5]
```diff
@@ -40,32 +47,6 @@ body {
 .sdds-btn-group sdds-button {
   margin: 10px;
 }
-/* TODO: remove lines 44-68 after release */
```
Comment (theJohnnyMe) [inline demo/react/src/index.scss:43]: You actually did delete - never mind my first comment. Good work ;)

**Comment by theJohnnyMe** [comment]
> _Side menu still moves with page, but I guess this is intentional._
> 
> Confirming that header is sticky on mobile and desktop with React, Angular 11, Angular 12 and Angular 13.
> 
> _Weird bug with Edge and Angular13._ Header sticky on "mobile" initially, but when I open side menu, scroll down, and then close side menu, header isn't sticky anymore. Doesn't go away when I run `npm i` and `npm start` or clean cache. Perhaps something wrong with Edge simulator for mobile devices?

I had an issue with extra space on both sides and top/bottom. I have removed the entire code from inspect tool, but those sliders to move the page up/down and left/right remained. Super wierd. I do not think this is related to our code.

**Review by theJohnnyMe — APPROVED** [review]
There is a part of the code that you commented out, see if you want to keep like that or delete it. Otherwise looks good to me.

File: demo/react/src/components/header/Header.css (hunk 4 above)
Comment (gaonita) [reply demo/react/src/components/header/Header.css:1]: yep I'm on it 👍🏻

**Comment by gaonita** [comment]
corrupted with old netlify checks to pass

**Pull request merged**